lib/core/src/events/ListenerBag.ts

```diff
@@ -78,6 +78,34 @@ export class ListenerBag {
     return this.listeners.length !== 0
   }
 
+  /**
+   * Checks if this listener bag instance has at least one of `name` assigned
+   *
+   * @param name The class name of the Listener to be queried
+   * @return Whether this instance has at least one of listener `name`
+   */
+  public hasListener (name: string) : boolean {
```

Comment:
Did you try to pass here something like `listener: Listener`? I assume that didn't work because `interface Listener` didn't have `listener.constructor` property?

Reply:
I may be totally wrong but I think the issue with this is that as we are adding instances to the  listener bag we would need a reference to the instance itself to pass to `hasListener()`, but normally the instance is created in `addListener()` as in: `addListener(new NameOfListener)`... We could maybe get around this if we used a mapping to store the listeners in `ListenerBag`, as we do for events in `Dispatcher` but then we would either be prevented from adding anything anonymous to `ListenerBag.listeners` or we would have to find a way to programmatically and systematically tag anonymous Hooks/Listeners for the mapping (or we could just ban anon entities from the ListenerBag and break a bunch of stuff in process).

As a result I felt that this was the way to query/retrieve/remove listener instances from the listeners array. One worry that I have is that our bundler/minification service could overwrite `<instance>.constructor.name` (see [here](https://stackoverflow.com/questions/10314338/get-name-of-object-or-class#comment17433297_10314492) and [here](https://stackoverflow.com/questions/46217853/get-constructor-name-of-object#comment79399855_46217896)) and then this would fail, so I will need to check on that and maybe add some note in the docs about issues with bundler configuration if relevant...

Reply:
👍  done, thanks @kouks